src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java

```diff
@@ -240,7 +242,14 @@ public void send(UnifiedMessage unifiedMessage) {
      * @param callback the {@link MessageResponseCallback} that will be called once the POST request completes.
      */
     private void submitPayload(String url, String jsonPayloadObject, String pushApplicationId, String masterSecret,
-            MessageResponseCallback callback) {
+            MessageResponseCallback callback, List<String> redirectUrls) {
+        if (redirectUrls.contains(url)) {
+            throw new IllegalStateException("The site contains an infinite redirect loop! Duplicate url: " +
+                    url);
+        } else {
+            redirectUrls.add(url);
```

Comment:
yes but if there is an redirect then we save an "useless" post by doing this early. Not sure what is the best.